### ISSUE TITLE
Exception: Invalid PBX project (parsing line 0)

### ISSUE THREAD

**Issue by sorab2142** [playgameservices/play-games-plugin-for-unity#1082]
Unity 5.3.4f1, IOS BUILD

Exception: Invalid PBX project (parsing line 0)
UnityEditor.iOS.Xcode.PBX.Parser.ParseTree () (at /Users/builduser/buildslave/unity/build/PlatformDependent/iPhonePlayer/Extensions/XCode/PBX/Parser.cs:114)
UnityEditor.iOS.Xcode.PBXProjectData.ParseContent (System.String content) (at /Users/builduser/buildslave/unity/build/PlatformDependent/iPhonePlayer/Extensions/XCode/PBXProjectData.cs:451)
UnityEditor.iOS.Xcode.PBXProjectData.ReadFromStream (System.IO.TextReader sr) (at /Users/builduser/buildslave/unity/build/PlatformDependent/iPhonePlayer/Extensions/XCode/PBXProjectData.cs:458)
UnityEditor.iOS.Xcode.PBXProject.ReadFromStream (System.IO.TextReader sr) (at /Users/builduser/buildslave/unity/build/PlatformDependent/iPhonePlayer/Extensions/XCode/PBXProject.cs:868)
UnityEditor.iOS.Xcode.PBXProject.ReadFromString (System.String src) (at /Users/builduser/buildslave/unity/build/PlatformDependent/iPhonePlayer/Extensions/XCode/PBXProject.cs:863)
GooglePlayGames.Editor.GPGSPostBuild.UpdateGeneratedPbxproj (System.String pbxprojPath) (at Assets/GooglePlayGames/Editor/GPGSPostBuild.cs:239)
GooglePlayGames.Editor.GPGSPostBuild.OnPostprocessBuild (BuildTarget target, System.String pathToBuiltProject) (at Assets/GooglePlayGames/Editor/GPGSPostBuild.cs:132)
UnityEditor.HostView:OnGUI()

Could be the cause of the GoogleSignIn.h not found.

**Comment by Hoffa25** [comment]
Same problem here! Stuck for 3 days untill I solved it now. Its the new Xcode 7.3 thats causing the problem. Just install 7.2.1 again and you should be fine!

**Comment by claywilkinson** [comment]
Hmm. That is really strange.  Maybe the file is not closed or something?  If anyone hits this again, can you cat the pbx file and see if it looks weird?

The plugin is trying to add some flags and files to the project.  I don't think it could be the cause of the header files not being found.  That would point to a Cocoapods issue.

**Comment by sorab2142** [comment]
@claywilkinson Check out https://gist.github.com/sorab2142/c493d6d45a5844174d6e21a0edcb7ff5

**Comment by claywilkinson** [comment]
Thanks - so this one did not work?  I'll take a look.

**Comment by sorab2142** [comment]
Yes, this one did not work.

On Tue, Apr 5, 2016 at 9:53 PM Clayton Wilkinson <notifications@github.com>
wrote:

> Thanks - so this one did not work? I'll take a look.
>
> —
> You are receiving this because you authored the thread.
>
> Reply to this email directly or view it on GitHub
> <https://github.com/playgameservices/play-games-plugin-for-unity/issues/1082#issuecomment-205882256>
>
-- 

Regards,
Sorab Pithawala

**Comment by claywilkinson** [comment]
OK - I think I found the problem.  Are you using cocoapods?  I am and I was able to reproduce the error.  The fix that worked for me was to change the order of post processing.

In Assets/GooglePlayGames/Editor/GPGSPostBuild.cs,  line 46, change it to include the order:
`        [PostProcessBuild (99999)]`

This allows all the other post processors to run, then runs cocoapods last.

I'll update the plugin in the next release (coming soon!).

**Comment by heyalistair** [comment]
Hey @claywilkinson, thanks for your help.

It seems that [PostProcessBuild (99999)] doesn't quite fix it for me. (Also, I had to use [PostProcessBuildAttribute (99999)]) It does help though; GPGS does trip up the other plugins. 
The error happens directly in the GPGSPostBuild's OnPostprocessBuild after cocoapods updates the xcproject file. I think that the xcproject switches formats after cocoapods is finished with it, and then GPGS throws the error. 

It also seems like there was an old duplicate issue that was talking about it this. I added more details there a couple of weeks ago: #725 Any suggestions?

Thanks again

**Comment by claywilkinson** [comment]
Cocoapods is definitely rewriting the pbxproject file as xml, which is technically OK, but Unity's API only supports the json-like syntax.

Do the other plugins have postprocess build steps?  can you add the attribute to them also? does that help?  Or making 99999 be 999999 ?  

The alternative is to not run pod update, and do that manually - maybe that is the best solution?

**Comment by heyalistair** [comment]
> Do the other plugins have postprocess build steps? can you add the attribute to them also? does that help? Or making 99999 be 999999 ?

Thanks for following up! Sorry, yes. Adding the attribute does clear up the issues that GPGS was having with other plugins. (I edited my previous comment to make my language clearer.)

However, I still have an issue **inside** the GPGS plugin when building for iOS. At line 127 of GPGSPostBuild.cs, cocoapods runs its update:
`w.UsingCocoaPod = CocoaPodHelper.Update(pathToBuiltProject);`
and then at line 132, we make a call to
`UpdateGeneratedPbxproj(pathToBuiltProject + "/Unity-iPhone.xcodeproj/project.pbxproj");`
which will read and parse the project file from the disk. Executing this function throws the error at the top of this issue ticket. I suspect it is because the format is changed and PBXProject does not know how to parse the XML format.

Does that clear things up?

**Comment by MildaGenius** [comment]
We solve this problem by installing xcproj.

`cd /tmp`
`git clone https://github.com/0xced/xcproj.git`
`cd xcproj`
`sudo xcodebuild -target xcproj install DSTROOT=/`

**Comment by heyalistair** [comment]
@MildaGenius Thanks! In my case, I already had that installed. I did try to reinstall it per your instructions and that did not fix it for me, but I'm glad to hear it might work for some others.

(@claywilkinson) However, I did just find something else that worked for me! I edited GPGSPostBuild.cs to ensure that the pod update was the last thing that executes in the PostBuild function. I really just moved three lines of down a couple of lines.

The ending of the PostBuild function now looks like this:
```
            //Copy the podfile into the project.
            string podfile = "Assets/GooglePlayGames/Editor/Podfile.txt";
            string destpodfile = pathToBuiltProject + "/Podfile";
            if (!System.IO.File.Exists(destpodfile))
            {
                FileUtil.CopyFileOrDirectory(podfile, destpodfile);
            }
            UnityEngine.Debug.Log("Adding URL Types for authentication using PlistBuddy.");

            UpdateGeneratedInfoPlistFile(pathToBuiltProject + "/Info.plist");
            UpdateGeneratedPbxproj(pathToBuiltProject + "/Unity-iPhone.xcodeproj/project.pbxproj");

            GPGSInstructionWindow w = EditorWindow.GetWindow<GPGSInstructionWindow>(
                true,
                "Building for IOS",
                true);
            w.minSize = new UnityEngine.Vector2(400, 300);
            w.UsingCocoaPod = CocoaPodHelper.Update(pathToBuiltProject);

        #endif
#endif
        }
```

And that worked for me!

**Comment by claywilkinson** [comment]
Great! Thanks.

**Comment by claywilkinson** [comment]
Changed the processing order to be 99999 and moved the code around as suggested.  This is now included in 0.9.33.
